### PR TITLE
Log a warning after 60 secs to remind the user to run code on all hos…

### PR DESCRIPTION
**Pull request by yashk2810**
…ts for Cloud TPU 1VM

### CODE REVIEW

File: jax/lib/xla_bridge.py [hunk 1]
```diff
@@ -145,6 +145,22 @@ def _make_tpu_driver_client():
   return tpu_driver_client.TpuBackend.create(worker=FLAGS.jax_backend_target)
 
 
+def tpu_client_timer_callback(timer: float):
+  def _log_warning():
+    warnings.warn('Did you run your code on all the hosts?')
```
Comment (skye) [inline jax/lib/xla_bridge.py:150]: Let's say something about the TPU runtime taking a while to start, and I also think we should link to https://jax.readthedocs.io/en/latest/multi_process.html. Here's my suggestion, feel free to reword:
`TPU backend initialization is taking more than {timer} seconds. Did you run your code on all TPU hosts? See https://jax.readthedocs.io/en/latest/multi_process.html for more information.`

File: jax/lib/xla_bridge.py [hunk 2]
```diff
@@ -145,6 +145,22 @@ def _make_tpu_driver_client():
   return tpu_driver_client.TpuBackend.create(worker=FLAGS.jax_backend_target)
 
 
+def tpu_client_timer_callback(timer: float):
```
Comment (skye) [inline jax/lib/xla_bridge.py:148]: nit: rename `timer` to `timer_secs` or `warning_secs` or something.

File: tests/xla_bridge_test.py [hunk 3]
```diff
@@ -56,6 +64,15 @@ def test_local_devices(self):
     with self.assertRaisesRegex(RuntimeError, "Unknown backend foo"):
       xb.local_devices(backend="foo")
 
+  @mock.patch('jax.lib.xla_client.make_tpu_client', side_effect=mock_tpu_client)
+  def test_timer_tpu_warning_1vm(self, _):
+    with warnings.catch_warnings(record=True) as w:
+      warnings.simplefilter('always')
+      xb.tpu_client_timer_callback(0.01)
```
Comment (skye) [inline tests/xla_bridge_test.py:71]: Maybe put a sleep here just in case? Not sure if this is necessary.

File: tests/xla_bridge_test.py [hunk 4]
```diff
@@ -56,6 +64,15 @@ def test_local_devices(self):
     with self.assertRaisesRegex(RuntimeError, "Unknown backend foo"):
       xb.local_devices(backend="foo")
 
+  @mock.patch('jax.lib.xla_client.make_tpu_client', side_effect=mock_tpu_client)
```
Comment (skye) [inline tests/xla_bridge_test.py:67]: I think this is the first use of mocks in jax, so cool! :D

File: tests/xla_bridge_test.py [hunk 5]
```diff
@@ -56,6 +64,15 @@ def test_local_devices(self):
...
+  def test_timer_tpu_warning_1vm(self, _):
+    with warnings.catch_warnings(record=True) as w:
+      warnings.simplefilter('always')
+      xb.tpu_client_timer_callback(0.01)
+      msg = str(w[-1].message)
+      self.assertLen(w, 1)
```
Comment (skye) [inline tests/xla_bridge_test.py:73]: nit: put this above the `w[-1]` access

**Review by skye — CHANGES_REQUESTED** [review]
Beautiful!

File: tests/xla_bridge_test.py (hunk 5 above)
Comment (yashk2810) [reply tests/xla_bridge_test.py:73]: Done

File: tests/xla_bridge_test.py (hunk 3 above)
Comment (yashk2810) [reply tests/xla_bridge_test.py:71]: Its not necessary. The time is so small that it runs perfectly everytime. I tested it a lot of times.

File: tests/xla_bridge_test.py (hunk 4 above)
Comment (yashk2810) [reply tests/xla_bridge_test.py:67]: Woohoo!

File: jax/lib/xla_bridge.py (hunk 2 above)
Comment (yashk2810) [reply jax/lib/xla_bridge.py:148]: Done

File: jax/lib/xla_bridge.py (hunk 1 above)
Comment (yashk2810) [reply jax/lib/xla_bridge.py:150]: Done

**Review by skye — APPROVED** [review]

**Pull request merged**